src/Cache/Psr16CacheClient.php

```diff
@@ -29,20 +30,23 @@ class Psr16CacheClient implements CacheInterface
      * @param ICredentialProvider $authProvider
      * @param int|null $defaultTtlSeconds
      * @param bool|null $throwExceptions
+     * @param string $cacheName
      */
     public function __construct(
         IConfiguration      $configuration,
         ICredentialProvider $authProvider,
         ?int                $defaultTtlSeconds,
-        ?bool               $throwExceptions = null
+        ?bool               $throwExceptions = null,
+        string              $cacheName = self::DEFAULT_CACHE_NAME
```

Comment:
I would make this nullable and default to null, then do `?? self::DEFAULT_CACHE_NAME` on line 48 so that callers can pass null to indicate they want the default.